lib/gitlab/ci/templates/jobs/docker/Kaniko.yml

```diff
@@ -51,3 +53,48 @@ kaniko_publish:
     - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
     - if: $CI_PIPELINE_SOURCE == "web"
     - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+
+# kaniko_publish_arm: Extends the standard Kaniko publish job for ARM-based architectures.
+# This job inherits all logic from `.kaniko_publish` and customizes execution by using a specific GitLab runner tag.
+# The tag `architecture-aarch64` ensures that the job runs on an ARM64 (AArch64) compatible GitLab runner.
+#
+# Use Case:
+# - When building Docker images for ARM-based environments such as AWS Graviton, Apple M1/M2 chips, or Jetson devices.
+# - Ensures proper architecture targeting during multi-arch image builds or ARM-specific deployments.
+kaniko_publish_arm:
+  stage: publish
+  extends: .kaniko_publish
+  tags:
+    - architecture-aarch64
+  rules:
+    - when: always
+
+# Standalone multiarch manifest job with its own variables
+multiarch_manifest_publish:
+  stage: publish
+  image: docker:24.0.5
+  services:
+    - docker:dind
+  variables:
+    DOCKER_TLS_CERTDIR: ""
+    AMD_IMAGE: "harbor.ctic-dev.com/platform-engineering/getting-started:0.1.2"
+    ARM_IMAGE: "harbor.ctic-dev.com/platform-engineering/getting-started:0.1.5"
+    MULTIARCH_IMAGE: "harbor.ctic-dev.com/platform-engineering/getting-started:0.1.5-multiarch"
+    HARBOR_URL: "https://harbor.ctic-dev.com"
+  before_script:
+    - echo "$HARBOR_ROBOT_PASSWORD" | docker login --username "$HARBOR_ROBOT_USERNAME" --password-stdin "$HARBOR_URL"
+  script:
+    - |
+      set -e
+      echo "🔧 Creating manifest from:"
+      echo " - $AMD_IMAGE"
+      echo " - $ARM_IMAGE"
+      echo "=> $MULTIARCH_IMAGE"
+
+      docker manifest create "$MULTIARCH_IMAGE" \
+        --amend "$AMD_IMAGE" \
+        --amend "$ARM_IMAGE"
+
+      docker manifest push "$MULTIARCH_IMAGE"
+  rules:
+    - when: always
```

Comment:
For the rules I'd just copy the rules from https://github.com/chesapeaketechnology/gitlab-templates/blob/release/4/lib/gitlab/ci/templates/jobs/docker/Kaniko.yml#L50 for consistency. 

```
  rules:
    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
    - if: $CI_PIPELINE_SOURCE == "web"
    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
```